### ISSUE TITLE
Übereinanderliegende Animationen werden transparent

### ISSUE THREAD

**Issue by OrianeJoublin** [JirkaDellOro/FUDGE_Story#24]
Wenn ich die Regenanimation abspiele (die aus einem semitransparenten Bild besteht) und über eine andere lege, werden beide Animationen im Bereich der Überlappung komplett transparent.

Code: https://github.com/OrianeJoublin/VisualNovel/blob/main/Template/Source/Scenes/S7A_SceneDogCityRain.ts

Laufende Novel: https://orianejoublin.github.io/VisualNovel/Template/index.html

Ich habe die anderen Szenen mal auskommentiert. Die Animation sollte hier also direkt in der zweiten Szene stattfinden, sobald sich das ganze aktualisiert hat.

![image](https://user-images.githubusercontent.com/92022973/147232472-0ac5523a-d720-4a78-b757-71b9420dde22.png)

![image](https://user-images.githubusercontent.com/92022973/147232877-5078436e-fd65-47fc-ba20-ff150b72d71f.png)

![image](https://user-images.githubusercontent.com/92022973/147233002-3e430d72-073d-4c15-8eb0-527d45046e29.png)

**Comment by JirkaDellOro** [comment]
Ok, schöne Sachen machst Du. 

Rein technisch kann ich mir vorstellen, woher der Effekt kommt. Der Regen und Vogel konkurrieren um die Pixel auf der gleichen Ebene. Du willst, dass der Vogel vor dem Regen erscheint oder der Regen auch auf dem Vogel zu sehen ist?

Generell müssen wir uns aber auch überlegen, ob der Begriff Character derzeit nicht etwas überstrapaziert wird, wenn damit bildschirmfüllender Regen dargestellt wird. Hier sollten wir uns mal mit Riem zusammensetzen und schauen, wie der Bedarf für solche Effekte am besten gedeckt werden soll. Grundsätzlich hat eine Location drei Ebenen: back, middle und front. In back werden die Hintergründe dargestellt, in middle streiten sich jetzt Vogel und Regen und front wird wahrscheinlich noch gar nicht benutzt. Derzeit sind dort allerdings auch nur statische Bilder vorgesehen. Es wäre auch hier denkbar, ebenso in back, Animationen laufen zu lassen.

Grundsätzlich, steht zudem aber bei FUDGE-Story auch die ganze Funktionalität von FUDGE offen. Man könnte also auch tatsächlich die einzelnen Samen oder Regentropfen fliegen lassen...

**Comment by OrianeJoublin** [comment]
Vielen Dank! Also am sinnvollsten wäre es, wenn der Regen auch auf dem Vogel zu sehen wäre. Aber ja, vielleicht wäre es sinnvoll über den Charakterbegriff zu sprechen und zu schauen, ob es für solche Animationen noch eine andere Kategorie geben sollte...

**Comment by JirkaDellOro** [comment]
Um meine These zu überprüfen, könntest Du bis dahin mal probieren, wie es sich darstellt, wenn Du den Vogel zuerst ins Spiel bringst und danach den Regen.

**Comment by Rina14** [comment]
Hab's grad mal getestet: Wenn der Vogel zuerst angezeigt wird, gibt es keine Überlappung.

@JirkaDellOro Bezüglich Locations und Animationen gibt es tatsächlich noch den Shaker, über den ich mal mit Dir reden wollte. Den hatte ich letztes Semester mit Robin eingebaut, um den Hintergrund zu animieren. Das würde sich hier ganz gut ergeben auch gleich darüber zu sprechen.

**Comment by OrianeJoublin** [comment]
Ich habe es jetzt auch endlich mal ausprobiert und hab dasselbe Ergebnis wie Riem: wenn der Vogel zuerst erscheint, wird er durch den Regen auch nicht durchsichtig.

**Comment by JirkaDellOro** [comment]
Ok, dann haben wir vielleicht erstmal einen Workaround und können uns in Ruhe Gedanken machen um eine bessere Struktur.

**Comment by OrianeJoublin** [comment]
Naja, der Vogel kommt eigentlich erst viel später in der Szene also kann ich ihn nicht vor dem Regen einfliegen lassen. Ich könnte höchstens eine der beiden Animationen ganz weglassen...

**Comment by JirkaDellOro** [comment]
Nein, Du kannst den Vogel schon außerhalb des sichtbaren Bereiches platzieren, denke ich und dann zur richtigen Zeit die Animation anwenden. Probier' mal bitte.

**Comment by OrianeJoublin** [comment]
Daran hab ich nicht gedacht. danke. Funktioniert!
